### PR TITLE
added tutorial for NeMo tokenizers

### PR DESCRIPTION
**Pull request by yzhang123**
Signed-off-by: Yang Zhang <yangzhang@nvidia.com>

### CODE REVIEW

**Review by ekmb — CHANGES_REQUESTED** [review]
@yzhang123, @titu1994 can we move https://github.com/NVIDIA/NeMo/blob/main/scripts/process_asr_text_tokenizer.py to common collection if it's the only example on how to train SPE?

@yzhang123 could you please add to your example on how to add an "'additional_special_tokens'".
This could be helpful https://github.com/NVIDIA/NeMo/blob/pm_nlg_on_sgd/examples/nlp/dialogue_state_tracking/dpm_nlg_sgd.py#L263

**Comment by titu1994** [comment]
@ekmb It's a script, so I don't think it should be a part of NeMo core. There's a lot more flexibility in leaving it as a script in order to provide it paths to manifest files.

**Comment by ekmb** [comment]
> @ekmb It's a script, so I don't think it should be a part of NeMo core. There's a lot more flexibility in leaving it as a script in order to provide it paths to manifest files.

It should stay as a script but shouldn't it be moved to collections/common/tokenizers if we're referring to it in the NLP tutorial?

**Comment by titu1994** [comment]
No I don't think that's good. Then users will need to hunt down a script inside Nemo collections. All scripts belong in the script folder only.

File: tutorials/nlp/02_NLP_Tokenizers.ipynb
Comment (VahidooX) [inline tutorials/nlp/02_NLP_Tokenizers.ipynb:1]: I think is is better not to use tokenizer.tokernizer in our code. Instead of that, it is better to user @property and @setter for such common properties as we encapsulate the tokenizer. We can get out those values from the tokenizer.tokenizer and pass it out. In the current design the internal tokenizers should all have the same properties which reduced our flexibility to support many tokenizers.

Comment (yzhang123) [reply tutorials/nlp/02_NLP_Tokenizers.ipynb:1]: This is just to showcase the wrapper for AutoTokenizer, so that when users want to use HuggingFace tokenizers they know how to access that. This notebook explores all possibilities with nemo tokenizers.

**Comment by yzhang123** [comment]
> @yzhang123 could you please add to your example on how to add an "'additional_special_tokens'".
> This could be helpful https://github.com/NVIDIA/NeMo/blob/pm_nlg_on_sgd/examples/nlp/dialogue_state_tracking/dpm_nlg_sgd.py#L263

I have not worked with that, so if you want to add it feel free.
But i think its always the best practice to specify special tokens in the constructor instead of calling it later

**Comment by yzhang123** [comment]
> No I don't think that's good. Then users will need to hunt down a script inside Nemo collections. All scripts belong in the script folder only.

Since I reuse @titu1994 script, I'll give @titu1994 the final call since it was already there. Also the script path is in the tutorial so i think its fine

**Comment by yzhang123** [comment]
> @yzhang123 could you please add to your example on how to add an "'additional_special_tokens'".
> This could be helpful https://github.com/NVIDIA/NeMo/blob/pm_nlg_on_sgd/examples/nlp/dialogue_state_tracking/dpm_nlg_sgd.py#L263

I have not used that in combination with resize embedding, so if you want feel free to add it. 
As far as adding special tokens, the notebook shows how to do that in the constructor. And that should be the best practice, since it could have side effects

File: tutorials/nlp/02_NLP_Tokenizers.ipynb
Comment (VahidooX) [reply tutorials/nlp/02_NLP_Tokenizers.ipynb:1]: But in this tutorial we are telling our users to use tokenzier.tokenzier.special_tokens to access the special tokens? How can user access the special_tokens without using tokenizer.tokenizer?

Comment (VahidooX) [reply tutorials/nlp/02_NLP_Tokenizers.ipynb:1]: If we do not want to add such feature in this PR, it is fine but I think we should add it to our TODO. We should not let the user to use tokenizer.tokenizer for common needs.

**Review by ekmb — APPROVED** [review]

**Review by titu1994 — APPROVED** [review]

**Comment by titu1994** [comment]
@yzhang123 Could you do one last refresh of the notebook from the colab before merge ?

**Pull request merged**
